### PR TITLE
Issue #2610220: Create a views argument plugin for filtering by active store

### PR DESCRIPTION
**Pull request by jazzdrive3**
This also adds a StoreCacheContext, so we can use is in the argument definition.

### CODE REVIEW

**Comment by jazzdrive3** [comment]
What is the process for reviewing these?

File: modules/store/src/Cache/Context/StoreCacheContext.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce_store\Cache\Context\StoreCacheContext.
+ */
```
Comment (borisson) [inline modules/store/src/Cache/Context/StoreCacheContext.php:6]: The `@file` docblocks for files that have only one class/interface can be removed, so let's remove this.

File: modules/store/src/Cache/Context/StoreCacheContext.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   *
+   * @var \Drupal\commerce_store\StoreContextInterface
+   */
+  protected $storeContext;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getLabel() {
+    return t('Store');
```
Comment (borisson) [inline modules/store/src/Cache/Context/StoreCacheContext.php:42]: Let's use the `StringTranslationTrait` and use `$this->t(` here instead if that's possible.

File: modules/store/src/Plugin/views/argument_default/ActiveStore.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce_store\Plugin\views\argument_default\ActiveStore.
+ */
```
Comment (borisson) [inline modules/store/src/Plugin/views/argument_default/ActiveStore.php:6]: The @file docblocks for files that have only one class/interface can be removed, so let's remove this.

**Comment by borisson** [comment]
I think this is a good addition, but it'll probably need a rebase and you should also squash the commits after resolving the nitpicks.

File: modules/store/src/Cache/Context/StoreCacheContext.php (hunk 2 above)
Comment (mglaman) [reply modules/store/src/Cache/Context/StoreCacheContext.php:42]: It's a static method

**Comment by bojanz** [comment]
Continued in https://www.drupal.org/node/2610220.

**Pull request closed without merging**